crates/api/src/peer_meta_store.rs

```diff
@@ -1,27 +1,48 @@
 use crate::{builder, config, BoxFut, K2Result, Timestamp, Url};
+use bytes::Bytes;
 use futures::future::BoxFuture;
 use std::sync::Arc;
 
 /// A store for peer metadata.
 ///
 /// This is expected to be backed by a key-value store that keys by space, peer URL and key.
 pub trait PeerMetaStore: 'static + Send + Sync + std::fmt::Debug {
-    /// Store a key-value pair for a given space and peer.
+    /// Store a key-value pair for a peer.
     fn put(
         &self,
         peer: Url,
         key: String,
-        value: bytes::Bytes,
+        value: Bytes,
         expiry: Option<Timestamp>,
     ) -> BoxFuture<'_, K2Result<()>>;
 
-    /// Get a value by key for a given space and peer.
+    /// Get a value by key for a peer.
     fn get(
         &self,
         peer: Url,
         key: String,
     ) -> BoxFuture<'_, K2Result<Option<bytes::Bytes>>>;
 
+    /// Mark a peer url unresponsive with an expiration timestamp.
+    //
+    // TODO After the expiry the peer url will be removed from the store.
+    fn mark_peer_unresponsive(
+        &self,
+        peer: Url,
+        expiry: Timestamp,
+    ) -> BoxFuture<'_, K2Result<()>> {
+        self.put(peer, "unresponsive".to_string(), Bytes::new(), Some(expiry))
```

Comment:
```suggestion
        self.put(peer, "transport:unresponsive".to_string(), Bytes::new(), Some(expiry))
```

Reply:
I hold that suggestion lightly. I just saw that in the gossip module the `gossip:` prefix was used and thought maybe we'd want to be consistent. But also, it's defined here in the `api` crate and not in the transport crate so maybe that prefix would not be appropriate anyway.